### PR TITLE
feat(behavior_path_planner): add geometric_parallel_parking

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Add parallel parkin planner used in pull_over (plan to use pull_out also).
Generate arc-based parking path using simple geometry of the vehicle kinematic model. 
image
(implementation of some part of [Geometric Path Planning for Automatic Parallel Parking in Tiny Spots](https://www.sciencedirect.com/science/article/pii/S1474667015347431))

## Related links

This is the part of https://github.com/autowarefoundation/autoware.universe/pull/873
depends on https://github.com/autowarefoundation/autoware.universe/pull/1257

## Tests performed

check pull_over module https://github.com/autowarefoundation/autoware.universe/pull/873 can work.

forward arc parking.
https://user-images.githubusercontent.com/39142679/177542524-e769692a-02e9-4a90-bb30-96095e6dbaa2.mp4

backward arc parking
https://user-images.githubusercontent.com/39142679/177542658-66c9cfde-2f88-484b-bc6e-af35f85a7e3e.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utilities.cpp [hunk 1]
```diff
@@ -792,8 +792,8 @@ PathWithLaneId removeOverlappingPoints(const PathWithLaneId & input_path)
       min_dist) {
       filtered_path.points.back().lane_ids.push_back(pt.lane_ids.front());
       filtered_path.points.back().point.longitudinal_velocity_mps = std::min(
-        pt.point.longitudinal_velocity_mps,
-        filtered_path.points.back().point.longitudinal_velocity_mps);
+        std::abs(pt.point.longitudinal_velocity_mps),
+        std::abs(filtered_path.points.back().point.longitudinal_velocity_mps));
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/utilities.cpp:796]: hmm. This seems to cause some problems

Comment (kosuke55) [reply planning/behavior_path_planner/src/utilities.cpp]: this is not needed so remove it. https://github.com/autowarefoundation/autoware.universe/pull/1260/commits/2625cad25ab5b05fd412af803a9092aead7bc86a

**Comment by kosuke55** [comment]
@yukkysaito 
moved it under scene_module/utils as `geometric_parallel_parking` https://github.com/autowarefoundation/autoware.universe/pull/1260/commits/7c85943b06f0945c6ad24b7f8fb0ff4c2c0bc29e

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**